### PR TITLE
Add AMBD platform support

### PR DESCRIPTION
**Pull request by AdamZhang0124**
#### Problem
Apply to support AMBD platform

#### Change overview
Added required files to add AMBD into the Device Layer
Added all-cluster-app example for AMBD
Added README file as the guide of the all-cluster-app example
Added build.sh to help on the compilation of the example

#### Testing
Tested all-cluster-app example with standalone chip-tool over WIFI.

### CODE REVIEW

File: examples/all-clusters-app/ambd/main/CHIPDeviceManager.cpp [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+CHIP_ERROR CHIPDeviceManager::Init(CHIPDeviceManagerCallbacks * cb)
+{
+    CHIP_ERROR err;
+    mCB = cb;
+
+    // Initialize the CHIP stack.
+    err = PlatformMgr().InitChipStack();
+    SuccessOrExit(err);
+
+    // switch (static_cast<RendezvousInformationFlags>(CONFIG_RENDEZVOUS_MODE))
```
Comment (andy31415) [inline examples/all-clusters-app/ambd/main/CHIPDeviceManager.cpp:64]: Remove commented out code

File: examples/all-clusters-app/ambd/main/CHIPDeviceManager.cpp [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+{
+    CHIPDeviceManagerCallbacks * cb = reinterpret_cast<CHIPDeviceManagerCallbacks *>(arg);
+    if (cb != nullptr)
+    {
+        printf("Common EventHandler hit!");
+        cb->DeviceEventCallback(event, reinterpret_cast<intptr_t>(cb));
+    }
+}
+
+/**
```
Comment (andy31415) [inline examples/all-clusters-app/ambd/main/CHIPDeviceManager.cpp:52]: remove empty comment

File: examples/all-clusters-app/ambd/main/CHIPDeviceManager.cpp [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+ *      This file implements the CHIP Device Interface that is used by
+ *      applications to interact with the CHIP stack
+ *
+ */
+
+#include <stdlib.h>
+
+#include "CHIPDeviceManager.h"
+#include <app/util/basic-types.h>
+//#include <setup_payload/SetupPayload.h>
```
Comment (andy31415) [inline examples/all-clusters-app/ambd/main/CHIPDeviceManager.cpp:29]: remove commented out code

**Review by andy31415 — COMMENTED** [review]
This PR contains a lot of commented out code. Please remove/replace with TODO description (while deleting the code. Do not add TOTO and still keep code 'as reference' as that has a great potential to become state) or implement as required.

File: examples/all-clusters-app/ambd/main/include/CHIPDeviceManager.h [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+     * @param event   ChipDeviceEvent that occurred
+     * @param arg     arguments specific to the event, if any
+     */
+    virtual void DeviceEventCallback(const chip::DeviceLayer::ChipDeviceEvent * event, intptr_t arg) {}
+
+    /**
+     * @brief
+     *   Called after an attribute has been changed
+     *
+     * @param endpoint           endpoint id
```
Comment (andy31415) [inline examples/all-clusters-app/ambd/main/include/CHIPDeviceManager.h:66]: delete redundant comments througout. It is clear that a variable of type chip::EndpointId and named 'endpoint' represents the 'endpoint id'.

Only keep comments that complement reading the code.

File: src/platform/AMBD/AMBDConfig.cpp [hunk 5]
```diff
@@ -0,0 +1,332 @@
+/*
+ *
+ *    Copyright (c) 2020 Project CHIP Authors
+ *    Copyright (c) 2019-2020 Google LLC.
+ *    Copyright (c) 2018 Nest Labs, Inc.
```
Comment (andy31415) [inline src/platform/AMBD/AMBDConfig.cpp:5]: Is this a new file or does it indeed need to keep the copyrights from nest and google?

**Comment by AdamZhang0124** [comment]
Hi @andy31415 , the redundant comments will be removed, thanks

File: src/platform/AMBD/AMBDConfig.cpp (hunk 5 above)
Comment (AdamZhang0124) [reply src/platform/AMBD/AMBDConfig.cpp:5]: It actually doesn't need the additional copyright. I was taking other platforms as reference.
Thanks for point it out, I will remove it.

**Comment by AdamZhang0124** [comment]
Hi @andy31415 , the redundant comments have been removed, please advice for the next step, thanks!

**Comment by woody-apple** [comment]
@AdamZhang0124 Thank you for your submission. Can you please sign the CLA? I've approved the PR run.


@mspang @saurabhst @msandstedt @Damian-Nordic @jmartinez-silabs @LuDuda can you review please? thanks!

**Review by woody-apple — APPROVED** [review]

File: examples/all-clusters-app/ambd/main/Makefile [hunk 6]
```diff
@@ -0,0 +1,207 @@
+#
```
Comment (jmartinez-silabs) [inline examples/all-clusters-app/ambd/main/Makefile:1]: What is the common opinion on the use of makefile ? We all migrate to GN or cmake in the beginning of Matter.

Comment (mspang) [reply examples/all-clusters-app/ambd/main/Makefile:1]: I don't think it's a good idea to use make here. The platform SDK already using it is one thing, but why add a brand new example using it?

**Pull request closed without merging**